lib/msal-common/src/authority/Authority.ts

```diff
@@ -543,17 +544,25 @@ export class Authority {
         let match = null;
         try {
             const response =
-                await this.networkInterface.sendGetRequestAsync<CloudInstanceDiscoveryResponse>(
+                await this.networkInterface.sendGetRequestAsync<CloudInstanceDiscoveryResponse | CloudInstanceDiscoveryErrorResponse>(
                     instanceDiscoveryEndpoint,
                     options
                 );
-            const metadata = isCloudInstanceDiscoveryResponse(response.body)
-                ? response.body.metadata
-                : [];
-            if (metadata.length === 0) {
-                // If no metadata is returned, authority is untrusted
-                return null;
+            
+            let typedResponseBody: CloudInstanceDiscoveryResponse | CloudInstanceDiscoveryErrorResponse;
+            let metadata: Array<CloudDiscoveryMetadata>;
+            if (isCloudInstanceDiscoveryResponse(response.body)) {
+                typedResponseBody = response.body as CloudInstanceDiscoveryResponse;
+                metadata = typedResponseBody.metadata;
+            } else {
+                typedResponseBody = response.body as CloudInstanceDiscoveryErrorResponse;
+                if (isCloudInstanceInvalid(typedResponseBody.error)) {
+                    return null;
+                }
+                
+                metadata = [];
```

Comment:
I'm not sure I understand, what is the difference here between null and empty array?

Reply:
On line 566, an array of metadata (cloud instances, can be empty) is needed to see if the developer's authority is within the cloud instances returned from the network request. And if there's no match... then apparently we make a new Authority for the purpose of a custom domain scenario, on line 576.

Reply:
It's probably to prserve behavior from 

```ts
const metadata = isCloudInstanceDiscoveryResponse(response.body)
                ? response.body.metadata
                : [];
```